src/figuresClasses.ts

```diff
@@ -1,19 +1,116 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape;
+
+  color: Color;
+
+  a: number;
+
+  b: number;
+
+  c: number;
 
+  constructor(color: Color, a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Length must be greater than 0.');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error(
+        'The longest side must be shorter than the sum of the other two sides',
+      );
+    }
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const halfPerimeter = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(
+      halfPerimeter
+      * (halfPerimeter - this.a)
+      * (halfPerimeter - this.b)
+      * (halfPerimeter - this.c),
+    );
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export class Circle {
+export class Circle implements Figure {
+  shape: Shape;
+
+  color: Color;
+
+  radius: number;
 
+  constructor(color: Color, radius: number) {
+    if (radius <= 0) {
+      throw new Error('Radius cannot be less than or equal to 0.');
+    }
+
+    this.shape = 'circle';
+    this.color = color;
+    this.radius = radius;
+  }
+
+  getArea(): number {
+    const area = Math.PI * this.radius * this.radius;
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Rectangle implements Figure {
+  shape: Shape;
+
+  color: Color;
+
+  width: number;
 
+  height: number;
+
+  constructor(color: Color, width: number, height: number) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('Width or height cannot be less than or equal to 0.');
+    }
+
+    this.shape = 'rectangle';
+    this.color = color;
+    this.width = width;
+    this.height = height;
+  }
+
+  getArea(): number {
+    const area = this.height * this.width;
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export function getInfo(figure: Figure): string {
+  let result: string = '';
+  const formattedString
+    = `A ${figure.color} ${figure.shape} - ${figure.getArea()}`;
+
+  if (figure.shape === 'triangle') {
```

Comment:
why do you need this?
<img width="800" alt="Screenshot 2023-05-17 at 16 28 28" src="https://github.com/mate-academy/ts_figures_classes/assets/53228259/99398a81-f844-4a6c-a7fd-a5cfd85e40a7">

you must return formattedString